CRADLE/CallPeak/callPeak.py

```diff
@@ -86,80 +98,69 @@ def mergePeaks(peakResult):
 			qvalues.extend([ currqvalue ])
 		else:
 			## update the continuous regions
-			minPValue = np.min(pvalues)
-			if(minPValue == 0):
-				mergedPeak[resultIdx][4] = np.nan
-			else:
-				mergedPeak[resultIdx][4] = np.round((-1) * np.log10(minPValue), 2)
-
-			minQValue = np.min(qvalues)
-			if(minQValue == 0):
-				mergedPeak[resultIdx][5] = np.nan
-			else:
-				mergedPeak[resultIdx][5] = np.round((-1) * np.log10(minQValue), 2)
+			mergedPeak[resultIdx][4] = updatePvalue(pvalues)
+			mergedPeak[resultIdx][5] = updateQvalue(qvalues)
 
 			regionChromo = mergedPeak[resultIdx][0]
 			regionStart = int(mergedPeak[resultIdx][1])
 			regionEnd = int(mergedPeak[resultIdx][2])
 
-			ctrlRC = []
-			for rep in range(vari.CTRLBW_NUM):
-				rc = np.nanmean(np.array(ctrlBW[rep].values(regionChromo, regionStart, regionEnd)))
-				ctrlRC.extend([rc])
-			ctrlRCPosMean = np.nanmean(ctrlRC)
+			ctrlRC, expRC = getRCFromBWs(ctrlBW, expBW, regionChromo, regionStart, regionEnd)
 
-			expRC = []
-			for rep in range(vari.EXPBW_NUM):
-				rc = np.nanmean(np.array(expBW[rep].values(regionChromo, regionStart, regionEnd)))
-				expRC.extend([rc])
+			ctrlRCPosMean = np.nanmean(ctrlRC)
 			expRCPosMean = np.nanmean(expRC)
-
 			diffPos = int(expRCPosMean - ctrlRCPosMean)
-			cohens_D = calculateCohenD(ctrlRC, expRC)
 			mergedPeak[resultIdx][6] = diffPos
+
+			cohens_D = calculateCohenD(ctrlRC, expRC)
 			mergedPeak[resultIdx].extend([ctrlRCPosMean, expRCPosMean, cohens_D])
+			
+			if vari.I_LOG2FC:
+				normCtrlRC, normExpRC = getRCFromBWs(normCtrlBW, normExpBW, regionChromo, regionStart, regionEnd)
+				normCtrlRCPosMean = np.nanmean(normCtrlRC)				
+				normExpRCPosMean = np.nanmean(normExpRC)
 
+				peusdoLog2FC = calculatePeusdoLog2FC(ctrlRCPosMean, expRCPosMean, normCtrlRCPosMean, normExpRCPosMean)
+			else:
+				peusdoLog2FC = np.nan
 
+			mergedPeak[resultIdx].extend([peusdoLog2FC])
+			
 			## start a new region
 			mergedPeak.append(peakResult[i])
 			pvalues = [currpvalue]
 			qvalues = [currqvalue]
 			resultIdx = resultIdx + 1
 
 		if i == (len(peakResult) -1):
-			minPValue = np.min(pvalues)
-			if(minPValue == 0):
-				mergedPeak[resultIdx][4] = np.nan
-			else:
-				mergedPeak[resultIdx][4] = np.round((-1) * np.log10(minPValue), 2)
-
-			minQValue = np.min(qvalues)
-			if(minQValue == 0):
-				mergedPeak[resultIdx][5] = np.nan
-			else:
-				mergedPeak[resultIdx][5] = np.round((-1) * np.log10(minQValue), 2)
+			mergedPeak[resultIdx][4] = updatePvalue(pvalues)
+			mergedPeak[resultIdx][5] = updateQvalue(qvalues)	
```

Comment:
For context, see later comment about these two functions
```suggestion
			mergedPeak[resultIdx][4] = newFC(pvalues)
			mergedPeak[resultIdx][5] = newFC(qvalues)	
```

Reply:
```suggestion
			mergedPeak[resultIdx][4] = takeMinusLog(pvalues)
			mergedPeak[resultIdx][5] = takeMinusLog(qvalues)	
```